pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2883,7 +2883,9 @@ public String startReplaceSegments(String tableNameWithType, List<String> segmen
           // 2. Proactively delete the oldest data snapshot to make sure that we only keep at most 2 data snapshots
           //    at any time in case of REFRESH use case.
           if (forceCleanup) {
-            if (lineageEntry.getState() == LineageEntryState.IN_PROGRESS && CollectionUtils
+            // TODO: Skip conflict checks for empty segmentsFrom for now. More conflict
+            //       checks over segmentsTo may be needed and added later.
+            if (lineageEntry.getState() == LineageEntryState.IN_PROGRESS && !segmentsFrom.isEmpty() && CollectionUtils
```

Comment:
We need to consider handling REFRESH & APPEND separately. For APPEND, we should not revert the `segmentsFrom=empty`; however, we need to revert for REFRESH.